Chessington.GameEngine/Pieces/Pawn.cs

```diff
@@ -14,11 +14,66 @@ public override IEnumerable<Square> GetAvailableMoves(Board board)
             
             if (this.Player == Player.White)
             {
-                availableMoves.Add(new Square(board.FindPiece(this).Row - 1, board.FindPiece(this).Col));
+                if (board.FindPiece(this).Row == 6 
+                    && board.GetPiece(new Square(board.FindPiece(this).Row - 2, board.FindPiece(this).Col)) == null 
+                    && board.GetPiece(new Square(board.FindPiece(this).Row - 1, board.FindPiece(this).Col)) == null)
+                { 
+                    availableMoves.Add(new Square(board.FindPiece(this).Row - 2, board.FindPiece(this).Col));
+                }
+                if (board.FindPiece(this).Row > 0)
+                {
+                    if (board.GetPiece(new Square(board.FindPiece(this).Row - 1, board.FindPiece(this).Col)) == null)
+                    {
+                        availableMoves.Add(new Square(board.FindPiece(this).Row - 1, board.FindPiece(this).Col));
+                    }
+ 
+                    if (board.FindPiece(this).Col + 1 < 7 && 
+                        board.GetPiece(new Square(board.FindPiece(this).Row - 1, board.FindPiece(this).Col + 1)) != null 
+                        &&  board.GetPiece(new Square(board.FindPiece(this).Row - 1, 
+                            board.FindPiece(this).Col + 1)).Player != this.Player)
```

Comment:
this sort of indentation is kind of hard to follow :D, i suggest using the same type of rule: for example here you have the first condition followed by the logical operator, then on one single line you got only a condition, then on the next two lines you got the logical operator first then a condition split because of function arguments, so it is kind of hard to follow

Reply:
That does make a lot of sense. Tried being consistent, but I guess I've missed some places. Also the lines could be shorter by using a variable for board.FindPiece(this) as you've mentioned earlier! Thank you